migtests/scripts/live-migration-fallf-run-test.sh

```diff
@@ -0,0 +1,181 @@
+	
+	step "START: ${TEST_NAME}"
+	print_env
+
+	pushd ${TEST_DIR}
+
+	step "Prepare FF metadata tables"
+	run_sqlplus_as_sys ${FF_DB_NAME} ${SCRIPTS}/oracle/fall-forward-prep.sql
+
+	step "Initialise source and fall forward database."
+	./init-db
+
+	step "Grant source database user permissions"
+	if [ "${SOURCE_DB_TYPE}" = "oracle" ]
+	then
+		grant_permissions_for_live_migration_oracle ${ORACLE_CDB_NAME} ${SOURCE_DB_NAME}
+	else
+		grant_permissions ${SOURCE_DB_NAME} ${SOURCE_DB_TYPE} ${SOURCE_DB_SCHEMA}
+	fi
+
+	step "Check the Voyager version installed"
+	yb-voyager version
+
+	step "Export schema."
+	export_schema
+	find ${EXPORT_DIR}/schema -name '*.sql' -printf "'%p'\n"| xargs grep -wh CREATE
+
+	step "Analyze schema."
+	analyze_schema
+	tail -20 ${EXPORT_DIR}/reports/report.txt
+
+	step "Fix schema."
+	if [ -x "${TEST_DIR}/fix-schema" ]
+	then
+		 "${TEST_DIR}/fix-schema"
+	fi
+
+	step "Analyze schema."
+	analyze_schema
+	tail -20 ${EXPORT_DIR}/reports/report.txt
+
+	step "Create target database."
+	run_ysql yugabyte "DROP DATABASE IF EXISTS ${TARGET_DB_NAME};"
+	run_ysql yugabyte "CREATE DATABASE ${TARGET_DB_NAME}"
+
+	step "Import schema."
+	import_schema
+	run_ysql ${TARGET_DB_NAME} "\dt"
+
+	step "Export data."
+	# false if exit code of export_data is non-zero
+	export_data --export-type "snapshot-and-changes" || { 
+		tail_log_file "yb-voyager-export-data.log"
+		tail_log_file "debezium-source_db_exporter.log"
+		exit 1
+	} &
+
+	# Storing the pid for the export data command
+	exp_pid=$!
+
+	# Killing the export process in case of failure
+	trap "kill_process -${exp_pid} ; exit 1" SIGINT SIGTERM EXIT SIGSEGV SIGHUP
+
+	# Waiting for snapshot to complete
+	timeout 100 bash -c -- 'while [ ! -f ${EXPORT_DIR}/metainfo/flags/exportDataDone ]; do sleep 3; done'
+
+	ls -l ${EXPORT_DIR}/data
+	cat ${EXPORT_DIR}/data/export_status.json || echo "No export_status.json found."
+	cat ${EXPORT_DIR}/metainfo/dataFileDescriptor.json 
+
+	step "Import data."
+	import_data --parallel-jobs 3 || { 
+		tail_log_file "yb-voyager-import-data.log"
+		exit 1
+	} &
+
+	# Storing the pid for the import data command
+	imp_pid=$!
+
+	# Updating the trap command to include the importer
+	trap "kill_process -${exp_pid} ; kill_process -${imp_pid} ; exit 1" SIGINT SIGTERM EXIT SIGSEGV SIGHUP
+
+	step "Fall Forward Setup"
+	fall_forward_setup || { 
+		tail_log_file "yb-voyager-fall-forward-setup.log"
+		exit 1
+	} &
+
+	# Storing the pid for the fall forward setup command
+	ffs_pid=$!
+
+	# Updating the trap command to include the ff setup
+	trap "kill_process -${exp_pid} ; kill_process -${imp_pid} ; kill_process -${ffs_pid} ; exit 1" SIGINT SIGTERM EXIT SIGSEGV SIGHUP
+
+	sleep 5m 
+
+	step "Run snapshot validations."
+	"${TEST_DIR}/validate"
+
+	step "Inserting new events to source"
+	run_sql_file source_delta.sql
+
+	sleep 2m
+
+	step "Initiating cutover"
+	yes | yb-voyager cutover initiate --export-dir ${EXPORT_DIR}
+
+	sleep 2m
```

Comment:
How about check for cutover status=COMPLETED? (with a timeout of 2m)

Reply:
yes agreed!